### DISCUSSION TITLE
How to provide context to context menu on right-clicking a Tab header?

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#15328] [Q&A]
I have a tabcontrol with a context menu:

```
      <TabControl.ContextMenu>
        <ContextMenu>
          <MenuItem Header="Test" Command="{Binding TestCommand}"/>
        </ContextMenu>
      </TabControl.ContextMenu>

```
When right-clicking on a not-selected tab header, I would like to provide the proper context to the menu, corresponding to the right-clicked header, whether selected or not. I don't want to select the header on opening the context menu.

This might be possible by specifying a command parameter in the axaml.
How do I specify the tab item as the command parameter?
{Binding} results in passing the outer datacontext:
```
      <TabControl.ContextMenu>
        <ContextMenu>
          <MenuItem Header="Test" Command="{Binding TestCommand}" CommandParameter="{Binding}" />
        </ContextMenu>
      </TabControl.ContextMenu>

```

**Comment by maxkatz6** [comment]
General approach with any control, not only TabControl, would be to find a specific visual element you want to have ContextFlyout on:
![image](https://github.com/AvaloniaUI/Avalonia/assets/3163374/508ffa14-22bb-48fa-8899-1e11420cd437)

Now you can write a selector like:
```
<Style Selector="TabControl#MyTabControl TabItem">
    <Setter Property="ContextMenu">
        <ContextMenu>
          <MenuItem Header="Test" Command="{Binding TestCommand}" CommandParameter="{Binding}" />
        </ContextMenu>
    </Setter>
</Style>
```

Keep in mind, TabItem will have data context of the page item, not the same as tabcontrol data context.

**Reply by wgraaf** [reply]
Thanks for the reply.
Where would I put the Style element?
I tried putting it in- and outside the TabControl, but I keep getting
'System.InvalidOperationException: 'Cannot use a control as a Setter value. Wrap the control in a <Template>.'

**Comment by Gawidev** [comment]
Managed to get this working for a ListBox ItemsControl with a context menu being defined in the parent window. This made it possible to bind via ReactiveUI code-behind which is the chosen binding method in my case.

In my window .axaml:
```xaml
<Window.Styles>
    <Style Selector="ListBox#MyListBox views|CustomItemDataTemplate">
        <Setter Property="ContextMenu">
            <ContextMenu>
                <MenuItem Name="CTXOpenFile" Header="Open File" />
                <MenuItem Name="CTXOpenLocation" Header="Open Location" />
                <MenuItem Name="CTXDeleteFile" Header="Delete File" />
            </ContextMenu>
        </Setter>
    </Style>
</Window.Styles>
```

Then in the code-behind constructor:
```cs
this.WhenActivated(compDispose =>
{
    this.BindCommand(ViewModel, vm => vm.OpenSelectedInDefaultProgram, v => v.CTXOpenFile)
         .DisposeWith(compDispose);
});
```

I knew about targeting controls by name, but wasn't aware it could be done in a Style Selector. Thanks for the info regardless, hopefully this example will be able to help others as well.
